packages/browser/src/sdk.ts

```diff
@@ -64,15 +64,27 @@ function applyDefaultOptions(optionsArg: BrowserOptions = {}): BrowserOptions {
     sendClientReports: true,
   };
 
-  // TODO: Instead of dropping just `defaultIntegrations`, we should simply
-  // call `dropUndefinedKeys` on the entire `optionsArg`.
-  // However, for this to work we need to adjust the `hasTracingEnabled()` logic
-  // first as it differentiates between `undefined` and the key not being in the object.
-  if (optionsArg.defaultIntegrations == null) {
-    delete optionsArg.defaultIntegrations;
+  return {
+    ...defaultOptions,
+    ...dropUndefinedKeysFlat(optionsArg),
+  };
+}
+
+/**
+ * In contrast to the regular `dropUndefinedKeys` method,
+ * this one does not deep-drop keys, but only on the top level.
+ */
+function dropUndefinedKeysFlat<T extends object>(obj: T): Partial<T> {
```

Comment:
Nitpick but I would rename this function to something like `dropTopLevelUndefinedKeys`, as `flat` is something I would not associate with the top-level.

And maybe we could add a test to check if top-level vs other-level keys are set as expected.